Fix: image isn't considered as a content appended to the news body so the post button is disabled - EXO-60812

hthabti opened the pull request:
prior to this change, when a user append an image in the body of the news during the creation or modification process, the post button is disabled.
after this change the post button is enabled when the appended content in the body is an image

halaya committed:
TASK-35704: Create FB maintenance and update projects versions/dependencies

halaya committed:
TASK-35704: Create FB maintenance and update projects versions/dependencies

halaya committed:
TASK-35704: Create FB maintenance and update projects versions/dependencies

halaya committed:
TASK-35704: Create FB maintenance and update projects versions/dependencies

halaya committed:
TASK-35704: Create FB maintenance and update projects versions/dependencies

akhanfir committed:
fix: News List template: use correct size for illustrations -EXO-60461 (#679)

Prior to this change, the size for illustrations in the templates lists news is still required at the original size of the image. After this change, the size of the illustrations is updated for the correct size.

akhanfir committed:
fix: News List template: use correct size for illustrations -EXO-60461 (#679)

Prior to this change, the size for illustrations in the templates lists news is still required at the original size of the image. After this change, the size of the illustrations is updated for the correct size.

hthabti committed:
image isn't considred as a content appended to the news body so the post button is disabled

ahamdi reviewed (changes requested):
When the image is inserted, the autoSave is fired in Create/Update article. I think that whenever we have a saved Draft, the Post button should be activated.
Could you change the logic of Post button, it does not have to rely on the content as it is HTML and could be escaped when transformed to article.

hthabti committed:
task-60812: image isn't considred as a content appended to the news body so the post button is disabled

put back the function getContent() as it was to not affect drafts autosave + made some edits on if conditionn of post button

hthabti commented:
@ahamdi I put back the line of the getcontent() as it was + made some edits on the if condition for postDisabled(). Thus saving drafts won't be affected and the button would be enabled when the content is an image

ahamdi committed:
Merge branch 'feature/maintenance' into task-60812

GouadriaHanen committed:
fix: Template News list images disabled case - EXO-61552 (#741)

In News-List template If images are off in template settings, no image should be displayed at all

GouadriaHanen committed:
fix: News Template admin/user view : Header is hidden if see all button if disabled - EXO-61514 (#743)

before this change, News Template admin/user view: Header, if see all button if disabled, is hidden by the article containers.
after this change, the header is not hidden after adding a min-height to its properties

hthabti committed:
Merge branch 'feature/maintenance' into task-60812

exo-swf commented:
Your PR triggers too many exo-ci builds! Please finish your work and then, set your PR ready! Thank you

exo-swf commented:
Your PR triggers too many exo-ci builds! Please finish your work and then, set your PR ready! Thank you

GouadriaHanen committed:
fix: Slider News templates : Center the Article title : EXO-61567 (#745)

prior to this change, for the user view the title of the news is not centered since the div of the setting bar is hidden
prior to this change, a margin-top is added to display the title centered

Jihed525 committed:
Merge branch 'feature/maintenance' into task-60812

ahamdi commented on webapp/src/main/webapp/news-activity-composer-app/components/ExoNewsActivityComposer.vue:
```diff
@@ -394,17 +394,24 @@ export default {
       }
       // disable update button if a mandatory field is empty
-      if (!this.news.title || !this.news.title.trim() || !this.news.body || !this.news.body.replace(/&nbsp;/g, '').trim()) {
+      if (!this.news.title || !this.news.title.trim() || !this.news.body.replace(/&nbsp;/g, '').trim() || new DOMParser().parseFromString(this.news.body, 'text/html').documentElement.querySelectorAll('img').length === new DOMParser().parseFromString(this.originalNews.body, 'text/html').documentElement.querySelectorAll('img').length && this.news.body === this.originalNews.body) {
```
we can not rely on the number of images, as we can add and remove an image in the same time and it won't be considered as a modification of an article

hthabti replied:
change done as follows
`      if (!this.news.title || !this.news.title.trim() || !this.news.body || !this.news.body.replace(/&nbsp;/g, '').trim()) {
        return true;
      }
      } else if (this.news.body.replace(/&nbsp;|<div>|<\/div>|<p>|<\/p>|\s+/g, '') !== this.originalNews.body.replace(/&nbsp;|<div>|<\/div>|<p>|<\/p>|\s+/g, '').replace(/\/>/g, '>').replace(/&#61;/g, '=')) {
        return false;
      } `

rdenarie committed:
fix : When an illustration is requested with a size, the image is always resized - EXO-61217

Before this fix, the cache is not well managed for news illustrations.
This fix improve etag usage and cache configuration so that the cache is better used, and the image less recalculated
In addition, in this fix, we return the correct image mimetype

rdenarie committed:
fix : When an illustration is requested with a size, the image is always resized - EXO-61217

Before this fix, the cache is not well managed for news illustrations.
This fix improve etag usage and cache configuration so that the cache is better used, and the image less recalculated
In addition, in this fix, we return the correct image mimetype

hthabti committed:
task-60812: relay on content to compare changes 

task-60812: relay on content to compare changes when editing an article

ahamdi committed:
Merge branch 'feature/maintenance' into task-60812

GouadriaHanen committed:
fix: for norml user the post button is disabled - EXO-61796 (#783)

before this change, for a simple user, the post button is disabled since no target is selected
after this change, the check of the selected target is only for those who can publish a news

hthabti committed:
Merge branch 'feature/maintenance' into task-60812

akhanfir committed:
fix: News template headlines+list: Move container to side layout output issue. - EXO-61647 (#787)

Prior this change, when move the headlines news template on side layout, The styles has changed : the title isn't extended on 2 lines, the space and news title on main article are in black. After this change , The same style in main container maintained.

akhanfir committed:
fix: News template headlines+list: Move container to side layout output issue. - EXO-61647 (#787)

Prior this change, when move the headlines news template on side layout, The styles has changed : the title isn't extended on 2 lines, the space and news title on main article are in black. After this change , The same style in main container maintained.

hthabti committed:
Merge branch 'feature/maintenance' into task-60812

akhanfir committed:
fix: News template headlines: Move container to side layout output issue. - EXO-61647 (#790)

Prior this change, when move the headlines news template on side layout, The styles has changed : the title isn't extended on 2 lines, the space and news title on main article are in black. After this change , The same style in main container maintained.

akhanfir committed:
fix: News template headlines: Move container to side layout output issue. - EXO-61647 (#790)

Prior this change, when move the headlines news template on side layout, The styles has changed : the title isn't extended on 2 lines, the space and news title on main article are in black. After this change , The same style in main container maintained.

Jihed525 committed:
Merge branch 'feature/maintenance' into task-60812

hthabti committed:
Merge branch 'feature/maintenance' into task-60812

hbenali commented:
Stale PR
